tests/tablespaces/Makefile

```diff
@@ -0,0 +1,49 @@
+PGVERSION ?= 10
+
+build:
+	$(MAKE) -C ../../ build-test-pg$(PGVERSION)
+	docker-compose build --build-arg "PGVERSION=$(PGVERSION)"
+
+teardown:
+	docker-compose down --volumes --remove-orphans
+
+tail:
+	docker-compose logs -f
+
+state:
+	docker-compose exec monitor pg_autoctl show state
+
+watch:
+	docker-compose exec monitor pg_autoctl watch
+
+test_01:
+	docker-compose up -d node1
+	TEST_NUM="01" docker-compose up --exit-code-from=test test
+
+test_02:
+	docker-compose up -d node2
+	TEST_NUM="02" docker-compose up --exit-code-from=test test
+
+test_03:
+	# Pollute the data dir so pg_rewind fails
+	docker-compose exec node2 mkdir '/var/lib/postgres/data/base/1/99999999'
+	docker-compose exec monitor pg_autoctl perform failover
```

Comment:
The split between the “infra” part in the Makefile, Dockerfile and docker-compose files and the “test/code” parts in the Python code is almost self-explanatory all along, apart from this bit here. Is there a strong reason why we would control the failover from the Makefile here rather than from the Python test file as usual?

If there is one, that's okay, it should be documented via a comment in the Makefile I think. It's not that trivial after all.

Reply:
Added some comments to the test files as well as the Makefile with some reasoning. 


I've been thinking about the other tests, if we wanted to move to a docker-compose approach:
We could have more python-test control if we added ssh access to the containers (this is non-trivial, but not terribly difficult, I believe). The caveat being stopping/starting the processes would require either a re-think of how the container commands run or external `docker-compose exec` like is being done here, since Docker will happily restart the container for you and there could be potential race conditions in your test (if it restarts it before you've done whatever work you wanted to do). 

Let me know if the updates are sufficient for this chunk of work!